shared/test-helpers/ganache-cli.sh

```diff
@@ -1,50 +1,58 @@
 #!/usr/bin/env bash
 
 # Exit script as soon as a command fails.
-set -o errexit
+set -o errexit -o pipefail
 
-if [ "$SOLIDITY_COVERAGE" = true ]; then
-  testrpc_port=8555
-else
-  testrpc_port=8545
-fi
+# Executes cleanup function at script exit.
+trap cleanup EXIT
 
-testrpc_running() {
-  nc -z localhost "$testrpc_port"
+cleanup() {
+  # Kill the RPC instance that we started (if we started one and if it's still running).
+  if [ -n "$rpc_pid" ] && ps -p $rpc_pid > /dev/null; then
+    kill -9 $rpc_pid
+  fi
 }
 
-start_testrpc() {
-  echo "Starting our own testrpc instance at port $testrpc_port"
-  if [ "$SOLIDITY_COVERAGE" = true ]; then
-    npx testrpc-sc -i 16 --gasLimit 0xfffffffffff --port "$testrpc_port"  > /dev/null &
-  else
-    npx ganache-cli -i 15 --gasLimit 50000000 --port "$testrpc_port" > /dev/null &
-  fi
+start_ganache() {
+  echo "Starting ganache-cli..."
+  npx ganache-cli -i ${NETWORK_ID} -l ${GAS_LIMIT} -e ${BALANCE} -p ${PORT} > /dev/null &
+  rpc_pid=$!
+  sleep 3
+  echo "Running ganache-cli with pid ${rpc_pid}"
+}
 
-  testrpc_pid=$!
+start_testrpc() {
+  echo "Starting testrpc-sc..."
+  npx testrpc-sc -i ${NETWORK_ID} -l ${GAS_LIMIT} -e ${BALANCE} -p ${PORT} > /dev/null &
+  rpc_pid=$!
+  sleep 3
+  echo "Running testrpc-sc with pid ${rpc_pid}"
 }
 
-if testrpc_running; then
-  echo "Killing testrpc instance at port $testrpc_port"
-  kill -9 $(lsof -i:$testrpc_port -t)
-fi
+measure_coverage() {
+  echo "Measuring coverage..."
+  npx solidity-coverage "$@"
+}
 
-start_testrpc
-sleep 5
+run_tests() {
+  echo "Running tests..."
+  npx truffle test --network rpc "$@"
+}
 
-# Exit error mode so the testrpc instance always gets killed
-set +e
-result=0
 if [ "$SOLIDITY_COVERAGE" = true ]; then
-  npx solidity-coverage "$@"
-  result=$?
-elif [ "$TRUFFLE_TEST" = true ]; then
-  truffle test --network rpc "$@"
-  result=$?
-fi
+  PORT=8555
+  BALANCE=10000
```

Comment:
Perhaps we should increase the balance here to be the same? No reason for the coverage account balance to be lower, I'd imagine 😄